### DISCUSSION TITLE
Homebrew update stuck

### DISCUSSION THREAD

**Discussion by bionicleFanatic** [Homebrew/discussions#3434] [Getting started] [answered]
Terribly sorry to bother everyone, but when running this in Terminal:
`brew update -d`
it seems to get stuck on this:
`++ [[ -z '' ]]
++ [[ -z '' ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ export HOMEBREW_UPDATE_TO_TAG=1
++ HOMEBREW_UPDATE_TO_TAG=1
++ [[ -z '' ]]
++ HOMEBREW_AUTO_UPDATE_SECS=60
++ [[ -e /usr/local/Cellar ]]
++ [[ ! -w /usr/local/Cellar ]]
++ [[ ! -w /usr/local/Homebrew ]]
++ git --version
++ export GIT_TERMINAL_PROMPT=0
++ GIT_TERMINAL_PROMPT=0
++ export 'GIT_SSH_COMMAND=ssh -oBatchMode=yes'
++ GIT_SSH_COMMAND='ssh -oBatchMode=yes'
++ [[ -z '' ]]
++ QUIET_ARGS=(-q)
++ unset GIT_CONFIG
++ lock update
++ local name=update
++ local lock_dir=/usr/local/var/homebrew/locks
++ local lock_file=/usr/local/var/homebrew/locks/update
++ [[ -d /usr/local/var/homebrew/locks ]]
++ [[ -w /usr/local/var/homebrew/locks ]]
++ exec
++ exec
++ _create_lock 200 update
++ local lock_fd=200
++ local name=update
++ local ruby=/usr/bin/ruby
++ [[ -x /usr/bin/ruby ]]
++ [[ -n /usr/bin/ruby ]]
++ /usr/bin/ruby -e 'File.new(200).flock(File::LOCK_EX | File::LOCK_NB) || exit(1)'
++ git_init_if_necessary
++ [[ -n 1 ]]
++ BREW_OFFICIAL_REMOTE=https://github.com/Homebrew/brew
++ CORE_OFFICIAL_REMOTE=https://github.com/Homebrew/homebrew-core
++ safe_cd /usr/local/Homebrew
++ cd /usr/local/Homebrew
++ [[ ! -d .git ]]
++ [[ -d /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core ]]
++ safe_cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
++ cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
++ [[ ! -d .git ]]
++ rename_taps_dir_if_necessary
++ local tap_dir
++ local tap_dir_basename
++ local tap_dir_hyphens
++ local user
++ local repo
++ for tap_dir in '"$HOMEBREW_LIBRARY"/Taps/*'
++ [[ -d /usr/local/Homebrew/Library/Taps/caskroom/.git ]]
++ continue
++ for tap_dir in '"$HOMEBREW_LIBRARY"/Taps/*'
++ [[ -d /usr/local/Homebrew/Library/Taps/homebrew/.git ]]
++ continue
++ safe_cd /usr/local/Homebrew
++ cd /usr/local/Homebrew
++ trap '{ /usr/bin/pkill -P $$; wait; exit 130; }' SIGINT
++ local update_failed_file=/usr/local/Homebrew/.git/UPDATE_FAILED
++ rm -f /usr/local/Homebrew/.git/UPDATE_FAILED
++ for DIR in '"$HOMEBREW_REPOSITORY"' '"$HOMEBREW_LIBRARY"/Taps/*/*'
++ [[ -d /usr/local/Homebrew/.git ]]
++ cd /usr/local/Homebrew
++ [[ -n '' ]]
+++ repo_var /usr/local/Homebrew
+++ local repo_var
+++ repo_var=/usr/local/Homebrew
+++ [[ /usr/local/Homebrew = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
+++ repo_var=
+++ echo ''
++ TAP_VAR=
+++ upstream_branch
+++ local upstream_branch
++++ git symbolic-ref refs/remotes/origin/HEAD
+++ upstream_branch=
+++ upstream_branch=
+++ [[ -z '' ]]
+++ upstream_branch=master
+++ echo master
++ UPSTREAM_BRANCH_DIR=master
++ declare UPSTREAM_BRANCH=master
+++ git rev-parse -q --verify refs/remotes/origin/master
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git rev-parse -q --verify refs/remotes/origin/master
++ declare PREFETCH_REVISION=0c785673d90ed080da50fcae0a85b15f6e038f72
++ [[ /usr/local/Homebrew = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
+++ git tag --list
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git tag --list
++ [[ -z 0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.7.1
0.8
0.8.1
0.9
0.9.1
0.9.2
0.9.3
0.9.4
0.9.5
0.9.8
0.9.9
1.0.0
1.0.1
1.0.2
1.0.3
1.0.4
1.0.5
1.0.6
1.0.7
1.0.8
1.0.9
1.1.0
1.1.1
1.1.10
1.1.11
1.1.12
1.1.13
1.1.2
1.1.3
1.1.4
1.1.5
1.1.6
1.1.7
1.1.8
1.1.9
1.2.0
1.2.1
1.2.2
1.2.3
1.2.4
1.2.5
1.2.6
1.3.0
1.3.1
1.3.2
1.3.3
1.3.4
1.3.5
1.3.6
1.3.7
1.3.8
1.3.9
1.4.0
1.4.1
1.4.2
1.4.3
1.5.0
1.5.1
1.5.10
1.5.11
1.5.12
1.5.13
1.5.14
1.5.2
1.5.3
1.5.4
1.5.5
1.5.6
1.5.7
1.5.8
1.5.9
1.6.0
1.6.1
1.6.10
1.6.11
1.6.12
1.6.13
1.6.14
1.6.15
1.6.16
1.6.17
1.6.2
1.6.3
1.6.4
1.6.5
1.6.6
1.6.7
1.6.8
1.6.9
1.7.0
1.7.1
1.7.2
1.7.3
1.7.4
1.7.5
1.7.6
1.7.7
1.8.0
1.8.1
1.8.2
1.8.3
1.8.4
1.8.5
1.8.6
1.9.0
1.9.1
1.9.2
1.9.3
2.0.0
2.0.1
2.0.2
2.0.3
2.0.4
2.0.5
2.0.6
2.1.0
2.1.1
2.1.10
2.1.11
2.1.12
2.1.13
2.1.14
2.1.15
2.1.16
2.1.2
2.1.3
2.1.4
2.1.5
2.1.6
2.1.7
2.1.8
2.1.9
2.2.0
2.2.1
2.2.10
2.2.11
2.2.12
2.2.13
2.2.14
2.2.15
2.2.16
2.2.17
2.2.2
2.2.3
2.2.4
2.2.5
2.2.6
2.2.7
2.2.8
2.2.9
2.3.0
2.4.0
2.4.1
2.4.10
2.4.11
2.4.12
2.4.13
2.4.14
2.4.15
2.4.16
2.4.2
2.4.3
2.4.4
2.4.5
2.4.6
2.4.7
2.4.8
2.4.9
2.5.0
2.5.1
2.5.10
2.5.11
2.5.12
2.5.2
2.5.3
2.5.4
2.5.5
2.5.6
2.5.7
2.5.8
2.5.9
2.6.0
2.6.1
2.6.2
2.7.0
2.7.1
2.7.2
2.7.3
2.7.4
2.7.5
2.7.6
2.7.7
3.0.0
3.0.1
3.0.10
3.0.11
3.0.2
3.0.3
3.0.4
3.0.5
3.0.6
3.0.7
3.0.8
3.0.9
3.1.0
3.1.1
3.1.10
3.1.11
3.1.12
3.1.2
3.1.3
3.1.4
3.1.5
3.1.6
3.1.7
3.1.8
3.1.9
3.2.0
3.2.1
3.2.10
3.2.11
3.2.12
3.2.13
3.2.14
3.2.15
3.2.16
3.2.17
3.2.2
3.2.3
3.2.4
3.2.5
3.2.6
3.2.7
3.2.8
3.2.9
3.3.0
3.3.1
3.3.10
3.3.11
3.3.12
3.3.13
3.3.14
3.3.15
3.3.16
3.3.2
3.3.3
3.3.4
3.3.5
3.3.6
3.3.7
3.3.8
3.3.9
3.4.0
3.4.1
3.4.10
3.4.11
3.4.2
3.4.3
3.4.4
3.4.5
3.4.6
3.4.7
3.4.8
3.4.9
3.5.0
3.5.1
3.5.2
3.5.3
3.5.4 ]]
++ [[ -z '' ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ for DIR in '"$HOMEBREW_REPOSITORY"' '"$HOMEBREW_LIBRARY"/Taps/*/*'
++ [[ -d /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask/.git ]]
++ cd /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
++ [[ -n '' ]]
++ [[ -n '' ]]
+++ repo_var /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
+++ local repo_var
+++ repo_var=/usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask
+++ [[ /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
+++ repo_var=/caskroom/homebrew-cask
+++ git config remote.origin.url
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git config remote.origin.url
++++ echo -n /caskroom/homebrew-cask
++++ tr -C A-Za-z0-9 _
++++ tr '[:lower:]' '[:upper:]'
++ UPSTREAM_REPOSITORY_URL=https://github.com/Homebrew/brew
++ [[ https://github.com/Homebrew/brew = \h\t\t\p\s\:\/\/\g\i\t\h\u\b\.\c\o\m\/* ]]
++ UPSTREAM_REPOSITORY=Homebrew/brew
++ UPSTREAM_REPOSITORY=Homebrew/brew
++ [[ /usr/local/Homebrew = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
++ [[ -n 1 ]]
+++ repo_var=_CASKROOM_HOMEBREW_CASK
+++ echo _CASKROOM_HOMEBREW_CASK
+++ sed -n 's/^ETag: "\([a-f0-9]\{32\}\)".*/\1/p' .git/GITHUB_HEADERS
++ TAP_VAR=_CASKROOM_HOMEBREW_CASK
+++ upstream_branch
+++ local upstream_branch
++++ git symbolic-ref refs/remotes/origin/HEAD
++ GITHUB_API_ETAG=
++ GITHUB_API_ACCEPT=application/vnd.github.v3+json
++ GITHUB_API_ENDPOINT=tags
+++ /usr/bin/curl --silent --max-time 3 --output /dev/null --write-out '%{http_code}' --dump-header /usr/local/Homebrew/.git/GITHUB_HEADERS --user-agent 'Homebrew/1.2.0 (Macintosh; Intel Mac OS X 10.12.6) curl/7.54.0' --header 'Accept: application/vnd.github.v3+json' --header 'If-None-Match: ""' https://api.github.com/repos/Homebrew/brew/tags
+++ upstream_branch=refs/remotes/origin/master
+++ upstream_branch=master
+++ [[ -z master ]]
+++ echo master
++ UPSTREAM_BRANCH_DIR=master
++ declare UPSTREAM_BRANCH_CASKROOM_HOMEBREW_CASK=master
+++ git rev-parse -q --verify refs/remotes/origin/master
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git rev-parse -q --verify refs/remotes/origin/master
++ declare PREFETCH_REVISION_CASKROOM_HOMEBREW_CASK=b0b1a9494c4911d80d440331a5ae2029c1844b29
++ [[ /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
++ [[ -z '' ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ for DIR in '"$HOMEBREW_REPOSITORY"' '"$HOMEBREW_LIBRARY"/Taps/*/*'
++ [[ -d /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/.git ]]
++ cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
++ [[ -n '' ]]
++ [[ -n '' ]]
+++ repo_var /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
+++ local repo_var
+++ repo_var=/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core
+++ [[ /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
+++ repo_var=/homebrew/homebrew-core
+++ git config remote.origin.url
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git config remote.origin.url
++++ echo -n /homebrew/homebrew-core
++++ tr -C A-Za-z0-9 _
++++ tr '[:lower:]' '[:upper:]'
++ UPSTREAM_REPOSITORY_URL=https://github.com/caskroom/homebrew-cask
++ [[ https://github.com/caskroom/homebrew-cask = \h\t\t\p\s\:\/\/\g\i\t\h\u\b\.\c\o\m\/* ]]
++ UPSTREAM_REPOSITORY=caskroom/homebrew-cask
++ UPSTREAM_REPOSITORY=caskroom/homebrew-cask
++ [[ /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
+++ git rev-parse refs/remotes/origin/master
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git rev-parse refs/remotes/origin/master
+++ repo_var=_HOMEBREW_HOMEBREW_CORE
+++ echo _HOMEBREW_HOMEBREW_CORE
++ TAP_VAR=_HOMEBREW_HOMEBREW_CORE
+++ upstream_branch
+++ local upstream_branch
++++ git symbolic-ref refs/remotes/origin/HEAD
++ GITHUB_API_ETAG=b0b1a9494c4911d80d440331a5ae2029c1844b29
++ GITHUB_API_ACCEPT=application/vnd.github.v3.sha
++ GITHUB_API_ENDPOINT=commits/master
+++ /usr/bin/curl --silent --max-time 3 --output /dev/null --write-out '%{http_code}' --dump-header /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask/.git/GITHUB_HEADERS --user-agent 'Homebrew/1.2.0 (Macintosh; Intel Mac OS X 10.12.6) curl/7.54.0' --header 'Accept: application/vnd.github.v3.sha' --header 'If-None-Match: "b0b1a9494c4911d80d440331a5ae2029c1844b29"' https://api.github.com/repos/caskroom/homebrew-cask/commits/master
+++ upstream_branch=refs/remotes/origin/master
+++ upstream_branch=master
+++ [[ -z master ]]
+++ echo master
++ UPSTREAM_BRANCH_DIR=master
++ declare UPSTREAM_BRANCH_HOMEBREW_HOMEBREW_CORE=master
+++ git rev-parse -q --verify refs/remotes/origin/master
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git rev-parse -q --verify refs/remotes/origin/master
++ declare PREFETCH_REVISION_HOMEBREW_HOMEBREW_CORE=2d9ae3ee42290f4b0c02ee08e3308fcbfd6aa057
++ [[ /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
++ [[ -z '' ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ wait
++ [[ -n '' ]]
+++ git config remote.origin.url
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git config remote.origin.url
++ UPSTREAM_REPOSITORY_URL=https://github.com/Homebrew/homebrew-core
++ [[ https://github.com/Homebrew/homebrew-core = \h\t\t\p\s\:\/\/\g\i\t\h\u\b\.\c\o\m\/* ]]
++ UPSTREAM_REPOSITORY=Homebrew/homebrew-core
++ UPSTREAM_REPOSITORY=Homebrew/homebrew-core
++ [[ /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core = \/\u\s\r\/\l\o\c\a\l\/\H\o\m\e\b\r\e\w ]]
+++ git rev-parse refs/remotes/origin/master
+++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
+++ /Applications/Xcode.app/Contents/Developer/usr/bin/git rev-parse refs/remotes/origin/master
++ GITHUB_API_ETAG=2d9ae3ee42290f4b0c02ee08e3308fcbfd6aa057
++ GITHUB_API_ACCEPT=application/vnd.github.v3.sha
++ GITHUB_API_ENDPOINT=commits/master
+++ /usr/bin/curl --silent --max-time 3 --output /dev/null --write-out '%{http_code}' --dump-header /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/.git/GITHUB_HEADERS --user-agent 'Homebrew/1.2.0 (Macintosh; Intel Mac OS X 10.12.6) curl/7.54.0' --header 'Accept: application/vnd.github.v3.sha' --header 'If-None-Match: "2d9ae3ee42290f4b0c02ee08e3308fcbfd6aa057"' https://api.github.com/repos/Homebrew/homebrew-core/commits/master
++ UPSTREAM_SHA_HTTP_CODE=200
++ [[ -f /usr/local/Homebrew/.git/FETCH_HEAD ]]
++ touch /usr/local/Homebrew/.git/FETCH_HEAD
++ UPSTREAM_SHA_HTTP_CODE=301
++ [[ -f /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask/.git/FETCH_HEAD ]]
++ touch /usr/local/Homebrew/Library/Taps/caskroom/homebrew-cask/.git/FETCH_HEAD
++ [[ -z '' ]]
++ [[ 200 = \3\0\4 ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ git fetch --tags --force -q origin refs/heads/master:refs/remotes/origin/master
++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
++ /Applications/Xcode.app/Contents/Developer/usr/bin/git fetch --tags --force -q origin refs/heads/master:refs/remotes/origin/master
++ [[ -z '' ]]
++ [[ 301 = \3\0\4 ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ git fetch --tags --force -q origin refs/heads/master:refs/remotes/origin/master
++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
++ /Applications/Xcode.app/Contents/Developer/usr/bin/git fetch --tags --force -q origin refs/heads/master:refs/remotes/origin/master
++ UPSTREAM_SHA_HTTP_CODE=200
++ [[ -f /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/.git/FETCH_HEAD ]]
++ touch /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/.git/FETCH_HEAD
++ [[ -z '' ]]
++ [[ 200 = \3\0\4 ]]
++ [[ -n '' ]]
++ [[ -n '' ]]
++ git fetch --tags --force -q origin refs/heads/master:refs/remotes/origin/master
++ [[ -z /Applications/Xcode.app/Contents/Developer/usr/bin/git ]]
++ /Applications/Xcode.app/Contents/Developer/usr/bin/git fetch --tags --force -q origin refs/heads/master:refs/remotes/origin/master`

From the look of it, it seems to not like something about Xcode. I'm using a mac, 10.12.6 Sierra. Any chance there's a workaround, or is this borked?

**Comment by p-linnane** [comment, marked as answer]
Did you run `brew update-reset && brew update`? Can you also please provide the output of `brew config` and `brew doctor --verbose`?

**Reply by bionicleFanatic** [reply]
That worked! Thanks very much :)
